src/lTerm_edit.ml

```diff
@@ -228,45 +227,16 @@ object(self)
     let line_count = Zed_lines.count line_set in
     let cursor_offset = Zed_cursor.get_position cursor in
     let cursor_line = Zed_lines.line_index line_set cursor_offset in
-    let line_start= Zed_lines.line_start line_set cursor_line in
     let cursor_column = cursor_offset - Zed_lines.line_start line_set cursor_line in
     let column_display= Zed_lines.force_width line_set (Zed_lines.line_start line_set cursor_line) cursor_column in
 
-    let calc_delta width_delta=
-      let rec forward idx acc=
-        if acc < width_delta then
-          forward (idx + 1) (acc + Zed_lines.force_width line_set idx 1)
-        else
-          idx
-      in
-      let rec backward idx acc=
-        if acc > width_delta then
-          backward (idx - 1) (acc - Zed_lines.force_width line_set (idx - 1) 1)
-        else
-          idx
-      in
-      if width_delta > 0 then
-        forward (line_start + shift) 0 - (line_start + shift)
-      else if width_delta < 0 then
-        backward cursor_offset 0 - cursor_offset
-      else 0
-    in
```

Comment:
There seems to be much less code now, does that translate to a difference in behaviour? It's just out of curiosity.

Reply:
The deleted code were written by me at a time when I start to prepare to upgrade lambda-term from 1.x to 2.x
At that time, for every component/part of the old code, we should determine which one should be ramained/updated/deleted and how the modifications will affect the backward-compatibility. Everything is obscure, uncertain.
At that time, I don't know if we can delete the `shift` value, which record the number of the characters for shifting, so I was wondering if we can add a `shift_width` and mix them together.
Now, I'm quite sure that `shift` is useless and almost all geometry-calculation should be based on the width of graphemes. So I deleted the useless code and simplify the related remaining code. And finally, implement the proper logic to support horizontal scrolling for wide graphemes.

Reply:
Sounds good, thanks for the explanation!